app/pages/gitlab_integration/project_page.rb

```diff
@@ -8,11 +8,23 @@ def initialize(current_user)
     end
 
     def build_project_names
-      @gitlab_client.load_repositories.map { |repo| [repo.namespace.name, repo.namespace.id] }.uniq
+      namespaces = @gitlab_client.load_repositories.map do |repo|
+        next if created_projects_integration_ids.include?(repo.namespace.id.to_s)
+
+        [repo.namespace.name, repo.namespace.id]
+      end
+
+      namespaces.compact.uniq
     end
 
     def namespace_by_id(integration_id)
       @gitlab_client.load_repositories.map(&:namespace).find { |namespace| namespace.id == integration_id.to_i }
     end
+
+    private
+
+    def created_projects_integration_ids
+      @_integration_ids ||= Project.where(integration_type: ::ProjectsConstants::Providers::GITLAB).pluck(:integration_id)
```

Comment:
Why you need memorized here?

Reply:
And why memorized variable is different from method name?

Reply:
Memoization is needed because of multiple method calls